pkg/build/controller/config_controller.go

```diff
@@ -6,19 +6,38 @@ import (
 	"github.com/golang/glog"
 	kapi "k8s.io/kubernetes/pkg/api"
 	kerrors "k8s.io/kubernetes/pkg/api/errors"
-	"k8s.io/kubernetes/pkg/util"
+	kutil "k8s.io/kubernetes/pkg/util"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
 	buildclient "github.com/openshift/origin/pkg/build/client"
+	"github.com/openshift/origin/pkg/build/util"
 )
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	BuildConfigDeleter      buildclient.BuildConfigDeleter
+	BuildDeleter            buildclient.BuildDeleter
+	BuildLister             buildclient.BuildLister
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if !bc.DeletionTimestamp.IsZero() {
```

Comment:
Relevant function in upstream is here https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/generic/etcd/etcd.go#L351 and another function called by it here https://github.com/kubernetes/kubernetes/blob/master/pkg/api/rest/delete.go#L42.

While researching when exactly the BC gets deleted, I discovered immediately. I'll look into it tomorrow. I think I might need to add that phase after all. :disappointed: